addons/stock/models/stock_inventory.py

```diff
@@ -164,6 +170,48 @@ def action_reset_product_qty(self):
         return True
 
     def action_validate(self):
+        # Check the serial numbers uniquiness
+        serials = self.line_ids.filtered(lambda l: l.product_id.tracking == 'serial' and l.prod_lot_id and l.theoretical_qty < l.product_qty)
+        wiz_lines = []
+        for line in serials:
+            similar_quant = self.env['stock.quant'].search([
+                ('lot_id', '=', line.prod_lot_id.id),
+                ('location_id', '!=', line.location_id.id),
+                '|',
+                ('location_id.usage', 'not in', ('supplier', 'inventory', 'production')),
+                ('location_id.scrap_location', '=', True),
+                ('quantity', '>', 0)
+            ])
+            # Extract the quant to correct
+            for sq in similar_quant:
+                related_lines = self.line_ids.filtered(
+                    lambda line: line.correction_line == 'warning' and
+                    line.prod_lot_id == sq.lot_id and
+                    line.location_id == sq.location_id and
+                    line.partner_id == sq.owner_id and
+                    line.package_id == sq.package_id
+                )
+                theoretical_qty = sum(related_lines.mapped('theoretical_qty'))
+                if float_compare(theoretical_qty, sq.quantity, precision_rounding=sq.product_id.uom_id.rounding) != 0:
+                    wiz_lines.append({
```

Comment:
you move a sn from shelf1 to shelf2.
line shelf1 theorical qty 1 qty 0
line shelf2 theorical qty 0 qty 1